src/Stack/BuildPlan.hs

```diff
@@ -735,7 +735,7 @@ data CustomSnapshot = CustomSnapshot
 instance FromJSON CustomSnapshot where
     parseJSON = withObject "CustomSnapshot" $ \o -> CustomSnapshot
         <$> ((o .: "compiler") >>= (\t -> maybe (fail $ "Invalid compiler: " ++ T.unpack t) return $ do
-                t' <- T.stripPrefix "ghc-" t
-                parseVersionFromString $ T.unpack t'))
+                GhcVersion v <- parseCompilerVersion t
```

Comment:
I'll make the change myself, it will also give me a chance to compile/test it locally.